test/jdk/sun/security/util/AlgorithmConstraints/DisabledAlgorithmPermits.java

```diff
@@ -0,0 +1,177 @@
+                    new TestCase(1, "XDH", true),
+                    new TestCase(1, "X25519", true),
+                    new TestCase(1, "X448", false),
+                    new TestCase(1, "Ed448", true));
+            case "XDH" ->
+                Arrays.asList(
+                    new TestCase("XDH", false),
+                    new TestCase("X25519", true),
+                    new TestCase("X448", true),
+                    new TestCase("Ed448", true),
+                    new TestCase(1, "XDH", false),
+                    new TestCase(1, "X25519", false),
+                    new TestCase(1, "X448", false),
+                    new TestCase(1, "Ed448", true));
+            default -> null;
+        };
+
+        Objects.requireNonNull(expected, "algorithm being tested " +
+            algorithm + " not in expected table");
+        System.out.println("---");
+        var dac = new DisabledAlgorithmConstraints("x");
+        System.out.println("disabled algorithms = " + Security.getProperty("x"));
+
+        // Using only testType 0, this tests that permit(Set<>, String, null)
+        // will check only the algorithm against the disabled list
+        expected.stream().filter(n->n.testType == 0).forEach(tc -> {
+            boolean r = dac.permits(Set.of(CryptoPrimitive.SIGNATURE),
+                    tc.testAlgo, null);
+            System.out.print("\tpermits(Set.of(CryptoPrimitive.SIGNATURE), \"" +
+                tc.testAlgo + "\", null): " + r + " : " );
+            if (r != tc.expected) {
+                System.out.println("failed.");
+                throw new AssertionError("failed.  Expected " +
+                    tc.expected);
+            }
+            System.out.println("pass");
+        });
+
+        // Using only testType 1, this tests permit(Set<>, Key) that will look
+        // at both the key.getAlgorithm() and the key.getParams().getName()
+        // against the disabled list
+        expected.stream().filter(n->n.testType == 1).forEach(tc -> {
+            PrivateKey k;
+            try {
+                k = KeyPairGenerator.getInstance(tc.testAlgo).generateKeyPair().
+                    getPrivate();
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+            boolean r = dac.permits(Set.of(CryptoPrimitive.SIGNATURE), k);
+            System.out.print("\tpermits(Set.of(CryptoPrimitive.SIGNATURE), " +
+                tc.testAlgo + " privkey): " + r + " : " );
+            if (r != tc.expected) {
+                System.out.println("failed.");
+                throw new AssertionError("failed.  Expected " +
+                    tc.expected);
+            }
+            System.out.println("pass");
+        });
+        System.out.println("---");
+    }
+
+    record TestCase(int testType, String testAlgo, boolean expected) {
+        TestCase(String testAlgo, boolean expected) {
+            this( 0, testAlgo, expected);
```

Comment:
Remove the space before '0'.

Reply:
ah.. intellij hid that from me

Reply:
You let it show the arg name?

Reply:
intellij put the variable name popup in the method line between the whitespace and the "0", so it wasn't obvious